analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+  dplyr::ungroup() |>
+  # add indicator columns for whether the cell is reference or query
+  dplyr::mutate(
+    immune_reference_cell = barcodes %in% immune_ref_barcodes,
+    subset_reference_cell = barcodes %in% subset_ref_barcodes
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df) |>
+  # finally, bring back in cluster and library information
+  dplyr::left_join(
+    infercnv_df |>
+      dplyr::select(barcodes, library_id, subcluster, normal_reference) |>
+      unique(),
+    by = c("barcodes", "normal_reference")
+  )
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+all_plot <- has_cnv_df |>
+  dplyr::filter(normal_reference == "all-immune") |>
+  ggplot() +
+  aes(x = total_cnv_per_cell, fill = immune_reference_cell) +
+  geom_density(alpha = 0.5) +
+  labs(
+    title = "Immune-all reference"
+  ) +
+  theme(legend.position = "bottom")
+
+subset_plot <- has_cnv_df |>
+  dplyr::filter(normal_reference == "subset-immune") |>
+  ggplot() +
+  aes(x = total_cnv_per_cell, fill = subset_reference_cell) +
+  geom_density(alpha = 0.5) +
+  labs(
+    title = "Immune-subset reference"
+  ) +
+  theme(legend.position = "bottom")
+
+wrap_plots(all_plot, subset_plot) +
+  plot_annotation(title = "Total CNVs per cell")
+```
+
+### CNVs across cell types
+
+Next, we'll look at the per-cell distributions of CNVs across cell types.
+Specifically, we'll look at both the cell types from the `cell-type-consensus` module and the `cell-type-ewings` module separately.
+
+Ideally, we'll see that Unknown cells from `cell-type-consensus` and cells labeled as tumor in `cell-type-ewings` will have relatively higher CNV scores compared to other cell types.
+In addition, comparing distributions between `all-immune` and `subset-immune` may be helpful to identify the optimal normal reference strategy.
+
+```{r fig.height=10, fig.width=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
+query_has_cnv_df <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id)
+
+consensus_violin <- ggplot(query_has_cnv_df) +
+  aes(
+    x = consensus_annotation,
+    y = total_cnv_per_cell,
+    fill = normal_reference
+  ) +
+  geom_violin(scale = "width") +
+  scale_fill_brewer(palette = "Pastel2") +
+  labs(title = "Annotations from cell-type-consensus") +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+ewings_violin <- ggplot(query_has_cnv_df) +
+  aes(
+    x = ewing_annotation,
+    y = total_cnv_per_cell,
+    fill = normal_reference
+  ) +
+  geom_violin(scale = "width") +
+  scale_fill_brewer(palette = "Pastel2") +
+  labs(title = "Annotations from cell-type-ewings") +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+wrap_plots(consensus_violin, ewings_violin, ncol = 1, guides = "collect") +
+  plot_annotation(title = "CNV scores across cell types") &
+  theme(legend.position = "bottom")
+```
+
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
```

Comment:
I think same thing here about showing tumor vs non-normal reference vs reference in the colors rather than the original cell type.